ok - i just reversed the checks. It fails on my local as expected. With civi v5.41.2.

```
$ phpunit --filter testSubmitPaylaterMembership web/modules/contrib/webform_civicrm/tests/src/FunctionalJavascript/ContributionPayLaterTest.php
PHPUnit 8.5.15 by Sebastian Bergmann and contributors.

Testing Drupal\Tests\webform_civicrm\FunctionalJavascript\ContributionPayLaterTest
E                                                                   1 / 1 (100%)

Time: 1.19 minutes, Memory: 16.00 MB

There was 1 error:

1) Drupal\Tests\webform_civicrm\FunctionalJavascript\ContributionPayLaterTest::testSubmitPaylaterMembership
Undefined index: join_date

/Users/jitendra/www/d9/web/modules/contrib/webform_civicrm/tests/src/FunctionalJavascript/ContributionPayLaterTest.php:66
/Users/jitendra/www/d9/vendor/phpunit/phpunit/src/Framework/TestResult.php:703

ERRORS!
Tests: 1, Assertions: 14, Errors: 1.
```

--- a/tests/src/FunctionalJavascript/ContributionPayLaterTest.php
+++ b/tests/src/FunctionalJavascript/ContributionPayLaterTest.php
@@ -13,7 +13,7 @@ use Drupal\webform\Entity\Webform;
 final class ContributionPayLaterTest extends WebformCivicrmTestBase {
 
   /**
-   * Test Membership submission using Pay Later option.
+   * Test Membership submission using the Pay Later option.
    */
   public function testSubmitPaylaterMembership() {
     $this->createMembershipType(20);

